File: sic-api/Entities/Db/DbMailConfig.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace sic_api.Entities.Db;

[Table("db_mail_config")]
public class DbMailConfig : BaseEntity
{
    [Required]
    [MaxLength(100)]
    [Column("config_name")]
    public string ConfigName { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("smtp_server")]
    public string SmtpServer { get; set; } = default!;

    [Required]
    [Column("smtp_port")]
    public int SmtpPort { get; set; }

    [Required]
    [MaxLength(320)]
    [Column("email_from")]
    public string EmailFrom { get; set; } = default!;

    [Required]
    [MaxLength(100)]
    [Column("username")]
    public string Username { get; set; } = default!;

    [Required]
    [MaxLength(500)]
    [Column("password")]
    public string Password { get; set; } = default!;

    [Column("enable_ssl")]
    public bool EnableSsl { get; set; } = false;

    [Column("sort_order")]
    public int SortOrder { get; set; } = 1;

    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    [Column("max_retry")]
    public int MaxRetry { get; set; } = 3;

    [MaxLength(500)]
    [Column("description")]
    public string? Description { get; set; }

    public ICollection<DbMailQueue>? MailQueues { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/db/DbMailConfig.java
package com.softinter.sicapi.entity.db;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "db_mail_config")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class DbMailConfig extends BaseEntity {

    @Column(name = "config_name", nullable = false, length = 100)
    private String configName;

    @Column(name = "smtp_server", nullable = false, length = 255)
    private String smtpServer;

    @Column(name = "smtp_port", nullable = false)
    private Integer smtpPort;

    @Column(name = "email_from", nullable = false, length = 320)
    private String emailFrom;

    @Column(name = "username", nullable = false, length = 100)
    private String username;

    @Column(name = "password", nullable = false, length = 500)
    private String password;

    @Column(name = "enable_ssl")
    private Boolean enableSsl = false;

    @Column(name = "sort_order")
    private Integer sortOrder = 1;

    @Column(name = "is_active")
    private Boolean isActive = false;

    @Column(name = "max_retry")
    private Integer maxRetry = 3;

    @Column(name = "description", length = 500)
    private String description;

    @OneToMany(mappedBy = "mailConfig") // assuming DbMailQueue has a field 'mailConfig'
    private List<DbMailQueue> mailQueues = new ArrayList<>();
}
